File: android/share/src/main/java/com/dreampany/share/ui/model/UserItem.java
package com.dreampany.share.ui.model;

import androidx.annotation.NonNull;
import android.view.View;
import android.widget.TextView;

import com.dreampany.framework.ui.model.BaseItem;
import com.dreampany.media.data.model.Media;
import com.dreampany.share.R;
import com.dreampany.share.data.model.User;

import org.jetbrains.annotations.NotNull;

import java.io.Serializable;
import java.util.List;

import butterknife.BindView;
import eu.davidea.flexibleadapter.FlexibleAdapter;
import eu.davidea.flexibleadapter.items.IFlexible;

/**
 * Created by Hawladar Roman on 9/4/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 */
public class UserItem extends BaseItem<User, UserItem.ViewHolder> {

    private boolean flagged;

    private UserItem(User item, int layoutId) {
        super(item, layoutId);
    }

    public static UserItem getItem(@NonNull User item) {
        return new UserItem(item, R.layout.item_user);
    }

    @Override
    public ViewHolder createViewHolder(View view, FlexibleAdapter<IFlexible> adapter) {
        return null;
    }

    @Override
    public void bindViewHolder(FlexibleAdapter<IFlexible> adapter, ViewHolder holder, int position, List<Object> payloads) {
        holder.bind(position, this);
    }

    @Override
    public boolean filter(Serializable constraint) {
        return false;
    }

    static class ViewHolder extends BaseItem.ViewHolder {

        ViewHolder(@NotNull View view, @NotNull FlexibleAdapter<IFlexible> adapter) {
            super(view, adapter);
        }

        void bind(int position, UserItem item) {

        }
    }
}

File: android/cast/src/main/kotlin/com/dreampany/cast/ui/model/UserItem.kt
package com.dreampany.cast.ui.model

import android.view.View
import android.widget.TextView
import androidx.annotation.LayoutRes
import androidx.recyclerview.widget.RecyclerView
import com.dreampany.cast.R
import com.dreampany.cast.data.model.User
import com.dreampany.frame.ui.model.BaseItem
import com.google.common.base.Objects
import eu.davidea.flexibleadapter.FlexibleAdapter
import eu.davidea.flexibleadapter.items.IFlexible
import kotlinx.android.synthetic.main.item_user.view.*
import java.io.Serializable

/**
 * Created by Roman-372 on 6/27/2019
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
class UserItem : BaseItem<User, UserItem.ViewHolder> {

    constructor(item: User, @LayoutRes layoutId: Int) : super(item, layoutId) {

    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val item = other as UserItem
        return Objects.equal(item.getItem(), getItem())
    }

    override fun createViewHolder(
        view: View,
        adapter: FlexibleAdapter<IFlexible<RecyclerView.ViewHolder>>
    ): ViewHolder {
        return UserViewHolder(view, adapter)
    }

    override fun bindViewHolder(
        adapter: FlexibleAdapter<IFlexible<RecyclerView.ViewHolder>>,
        holder: ViewHolder,
        position: Int,
        payloads: MutableList<Any>
    ) {

    }

    override fun filter(constraint: Serializable): Boolean {
        return item.name.toLowerCase().startsWith((constraint as String).toLowerCase())
    }

    companion object {
        fun getUser(item: User): UserItem {
            return UserItem(item, R.layout.item_user)
        }
    }

    abstract class ViewHolder : BaseItem.ViewHolder {

        constructor(
            view: View,
            adapter: FlexibleAdapter<IFlexible<RecyclerView.ViewHolder>>
        ) : super(view, adapter) {

        }
    }

    class UserViewHolder : ViewHolder {

        lateinit var name : TextView

        constructor(
            view: View,
            adapter: FlexibleAdapter<IFlexible<RecyclerView.ViewHolder>>
        ) : super(view, adapter) {
            name = view.text_name
        }
    }
}